Source/engine.h

```diff
@@ -60,6 +60,26 @@ enum direction : uint8_t {
 struct Point {
 	int x;
 	int y;
+
+	Point operator +(const Point &other)
+	{
+		return {x + other.x, y + other.y};
+	}
+
+	Point operator -(const Point &other)
+	{
+		return {x - other.x, y - other.y};
+	}
+
+	void operator +=(const Point &other)
+	{
+		*this = *this + other;

```

Comment:
nit: Usually it's implemented the other way around (`+` in terms of `+=` and copy)

Reply:
I see no point doing it the other way around in this case. (Pun intended.)

Reply:
There is no advantage to either approach in this case, so please use the common approach for consistency. There is no need for this one to be special.